pybind_interface/pybind_main.cpp

```diff
@@ -375,6 +389,59 @@ std::vector<std::complex<float>> qsim_simulate(const py::dict &options) {
   return amplitudes;
 }
 
+std::vector<std::complex<float>> qtrajectory_simulate(const py::dict &options) {
+  NoisyCircuit<Cirq::GateCirq<float>> ncircuit;
+  unsigned num_qubits;
+  std::vector<Bitstring> bitstrings;
+  try {
+    ncircuit = getNoisyCircuit(options);
+    num_qubits = qsim::count_qubits(ncircuit);
+    bitstrings = getBitstrings(options, num_qubits);
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  using Simulator = qsim::Simulator<For>;
+  using StateSpace = Simulator::StateSpace;
+  using State = StateSpace::State;
+
+  // Define container for amplitudes
+  std::vector<std::complex<float>> amplitudes;
+  amplitudes.reserve(bitstrings.size());
```

Comment:
Is this reserve always for size zero ?

Reply:
No - `bitstrings` contains the list of bitstrings for which the user has requested amplitudes. This reserves an equal number of spaces for storing amplitudes.